bugsnag/client.py

```diff
@@ -226,6 +228,15 @@ def leave_breadcrumb(
             # to "MANUAL" if the given type is invalid
             type = BreadcrumbType.from_string(type)
 
+        if not isinstance(metadata, dict):
+            warning_message = 'breadcrumb metadata must be a dict, got {}'
+            warnings.warn(
+                warning_message.format(builtins.type(metadata).__name__),
+                RuntimeWarning
+            )
+
+            metadata = {}
```

Comment:
This matches the behaviour of JavaScript when something invalid is passed as breadcrumb metadata. Not sure if it's worth keeping the value (e.g. `metadata = {'?': metadata}`)

Reply:
I think dropping it is reasonable (and might even be specified somewhere?)